pkg/daemons/control/server.go

```diff
@@ -109,8 +109,12 @@ func controllerManager(cfg *config.Control, runtime *config.ControlRuntime) erro
 		"bind-address":                     localhostIP.String(),
 		"secure-port":                      "0",
 		"use-service-account-credentials":  "true",
-		"cluster-signing-cert-file":        runtime.ClientCA,
-		"cluster-signing-key-file":         runtime.ClientCAKey,
```

Comment:
It looks like removing this arg will break the current behavior of the legacy-unknown signer; can you verify the behavior with the proposed new flags? Are there dedicated flags for this signer?

Reply:
There are parameters for the legacy-unknown signer:
```
--cluster-signing-legacy-unknown-cert-file string          Filename containing a PEM-encoded X509 CA certificate used to issue certificates for the kubernetes.io/legacy-unknown signer.  If specified, --cluster-signing-{cert,key}-file must not be set.
--cluster-signing-legacy-unknown-key-file string           Filename containing a PEM-encoded RSA or ECDSA private key used to sign certificates for the kubernetes.io/legacy-unknown signer.  If specified, --cluster-signing-{cert,key}-file must not be set.
```
Which CA should be used?

Reply:
Great. I think that accounts for all the built-in signers? Are there any other signer args we're not setting that were defaulted by the previous args?

Reply:
I checked the source code of the kube-manager-controller to be sure. The specific parameters were introduced in v1.19.0 ([CHANGELOG/CHANGELOG-1.19.md](https://github.com/kubernetes/kubernetes/blob/v1.21.0/CHANGELOG/CHANGELOG-1.19.md#api-change-2)).

There are four built-in signer controllers ([pkg/controller/certificates/signer/signer.go](https://github.com/kubernetes/kubernetes/blob/v1.21.0/pkg/controller/certificates/signer/signer.go#L45-L79)):
* `csrsigning-kubelet-serving`
* `csrsigning-kubelet-client`
* `csrsigning-kube-apiserver-client`
* `csrsigning-legacy-unknown`

There is a parameter pair for each of these signers ([cmd/kube-controller-manager/app/options/csrsigningcontroller.go](https://github.com/kubernetes/kubernetes/blob/v1.21.0/cmd/kube-controller-manager/app/options/csrsigningcontroller.go#L40-L47)).

The single signing file (`--cluster-signing-cert-file`) is used as a fallback if the specific files are not set ([cmd/kube-controller-manager/app/certificates.go](https://github.com/kubernetes/kubernetes/blob/v1.21.0/cmd/kube-controller-manager/app/certificates.go#L55-L93)). I found no other usage of the single signing file.

With the new parameters the behavior stays the same except for the kubelet-serving signer which is now correctly signed with the server CA.